### DISCUSSION TITLE
Native Messaging distribution

### DISCUSSION THREAD

**Discussion by beriberikix** [wxt-dev/wxt#508] [Ideas]
Developing extensions that rely on Native Messaging brings a fair bit of complexity:

* Need to develop an installer (cross-platform)
* Need to develop/maintain an additional manifest (also xPlatform)
* Need to set OS configurations (ex. Registry on windows)

It would be awesome if there was additional tooling to help with this!

**Comment by aklinker1** [comment]
I don't know anything about the native side of native messaging... Is there a full, working example I could look at?

This seems more like the desktop app's responsibility. Could you explain more why you think this is something WXT should be responsible for and what benefits it would bring?

**Reply by beriberikix** [reply]
Here are more details: https://developer.chrome.com/docs/extensions/develop/concepts/native-messaging, https://github.com/GoogleChrome/chrome-extensions-samples/tree/main/api-samples/nativeMessaging/extension

Agree, this is the responsibility of WXT, but there's certainly opportunity to create tooling to _help_. Maybe it's just an example, but I can imagine a solution would be some combination of WXT + a cross-platform tool like [Tauri](https://tauri.app/).

**Comment by dylan1951** [comment]
Yes this would be really awesome.
